service/storage_service.cc

```diff
@@ -421,10 +421,11 @@ future<> storage_service::topology_state_load() {
                 // for the node were not in the 'normal' state yet
                 co_await update_peer_info(ip);
                 // And then amend with the info from raft
-                co_await _sys_ks.local().update_tokens(ip, id.uuid(), rs.ring.value().tokens);
                 co_await _sys_ks.local().update_peer_info(ip, "data_center", rs.datacenter);
                 co_await _sys_ks.local().update_peer_info(ip, "rack", rs.rack);
                 co_await _sys_ks.local().update_peer_info(ip, "release_version", rs.release_version);
+                // Finally, "seal" the state with host_id and tokens
```

Comment:
Hmm... we call `update_peer_info(ip)` at the beginning of this sequence. So the row actually gets host ID written before that, at least if gossiper is populated when we do this. Which means we don't achieve the goal explained in the commit message :(

Reply:
I'll look into accumulating the changes and then applying them in a single cql query.

Reply:
done in v2